cmd/osbuild-composer/composer.go

```diff
@@ -311,6 +315,25 @@ func (c *Composer) Start() error {
 				panic(err)
 			}
 		}()
+
+		if c.promListener != nil {
+			// metrics listener on another port
+			metricsMux := http.NewServeMux()
+			metricsMux.Handle("/metrics", promhttp.Handler().(http.HandlerFunc))
+
+			promServer = &http.Server{
+				ErrorLog:          c.logger,
+				Handler:           metricsMux,
+				ReadHeaderTimeout: 5 * time.Second,
+			}
+
+			go func() {
+				err := promServer.Serve(c.promListener)
+				if err != nil && err != http.ErrServerClosed {
+					panic(err)
+				}
+			}()
+		}
```

Comment:
I thought that this block will be moved outside of the `if c.apiListener != nil` block. 🤔